src/fusion.cpp

```diff
@@ -1388,51 +468,21 @@ miopenStatus_t FusionPlanDescriptor::Execute(const Handle& handle,
     {
         MIOPEN_THROW(miopenStatusBadParm, "The input descriptors dont match.");
     }
-
-    auto ops_head = op_map[0];
-
-    auto&& kernels = handle.GetKernels(algorithm_name, network_config);
-    MIOPEN_LOG_I(algorithm_name << ',' << network_config);
-    if(kernels.empty())
+    const auto& solution = solutions[0];
+    if(!solution.Succeeded())
     {
-        MIOPEN_THROW(miopenStatusBadParm, "The FusionPlan was not compiled for execution");
+        MIOPEN_THROW(miopenStatusBadParm, "The Fusion Plan was not compiled");
     }
-    KernelInvoke kernel = kernels.front();
 
-    std::vector<OpKernelArg> args;
-    if(arg_list.empty())
-    {
-        MIOPEN_THROW("Kernel arguments not setup properly");
-    }
-    for(auto& arg : arg_list)
-    {
-        MIOPEN_LOG_I2("Key: " + arg.key);
-        switch(arg.type)
-        {
-        case Input_Ptr: args.emplace_back(OpKernelArg(input)); break;
-        case Output_Ptr: args.emplace_back(OpKernelArg(output)); break;
-        case Padding: args.emplace_back(OpKernelArg(0, arg.size)); break;
-        case Scalar:
-        case Pointer: {
-            auto it = op_args.args_map.find(arg.key);
-            if(it != op_args.args_map.end())
-            {
-                args.push_back(it->second);
-            }
-            else
-            {
-                MIOPEN_THROW(miopenStatusInternalError, "Argument Not Set: " + arg.key);
-            }
-            break;
-        }
-        case Default: args.push_back(arg.val); break;
-        }
-    }
-    if(args.empty())
-    {
-        MIOPEN_THROW("Operator args not populated properly");
-    }
-    kernel(args);
+    const auto invoker = handle.GetInvoker(network_config, solver::Id{solution.solver_id}, {});
+    const auto& tmp    = op_args.params[1];
+    const auto& tmp2   = std::dynamic_pointer_cast<fusion::ActivationOpInvokeParam>(tmp);
+    const auto tmp3    = op_args.params[1];
```

Comment:
Are all this values not used at all? Is there any reason to create one set of "refs" (tmp2 is not really a ref as it owns the data it gets initialized from) and one of copies?

Reply:
This was redundant code and has been removed